lib/matplotlib/_constrained_layout.py

```diff
@@ -82,181 +73,361 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
+        margin['bottom'] += pos.y0 - bbox.y0
+        margin['top'] += -pos.y1 + bbox.y1
+
+        # make margin for colorbars.  These margins go in the
+        # padding margin, versus the margin for axes decorators.
+        for cbax in ax._colorbars:
+            # note pad is a fraction of the parent width...
+            pad = _colorbar_get_pad(cbax)
+            # colorbars can be child of more than one subplot spec:
+            cbp_rspan, cbp_cspan = _get_cb_parent_spans(cbax)
+            loc = cbax._colorbar_info['location']
+            cbpos, cbbbox = _get_pos_and_bbox(cbax, renderer)
+            if loc == 'right':
+                if cbp_cspan.stop == ss.colspan.stop:
+                    # only increase if the colorbar is on the right edge
+                    margin['rightcb'] += cbbbox.width + pad
+            elif loc == 'left':
+                if cbp_cspan.start == ss.colspan.start:
+                    # only increase if the colorbar is on the left edge
+                    margin['leftcb'] += cbbbox.width + pad
+            elif loc == 'top':
+                if cbp_rspan.start == ss.rowspan.start:
+                    margin['topcb'] += cbbbox.height + pad
+            else:
+                if cbp_rspan.stop == ss.rowspan.stop:
+                    margin['bottomcb'] += cbbbox.height + pad
+            # If the colorbars are wider than the parent box in the
+            # cross direction
+            if loc in ['top', 'bottom']:
+                if (cbp_cspan.start == ss.colspan.start and
+                        cbbbox.x0 < bbox.x0):
+                    margin['left'] += -cbbbox.x0 + bbox.x0
+                if (cbp_cspan.stop == ss.colspan.stop and
+                        cbbbox.x1 > bbox.x1):
+                    margin['right'] += cbbbox.x1 - bbox.x1
+            # or taller:
+            if loc in ['left', 'right']:
+                if (cbp_rspan.stop == ss.rowspan.stop and
+                        cbbbox.y0 < bbox.y0):
+                    margin['bottom'] += bbox.y0 - cbbbox.y0
+                if (cbp_rspan.start == ss.rowspan.start and
+                        cbbbox.y1 > bbox.y1):
+                    margin['top'] += -bbox.y1 + cbbbox.y1
+        # pass the new margins down to the layout grid for the solution...
+        gs._layoutgrid.edit_outer_margin_mins(margin, ss)
+
+
+def _make_margin_suptitles(fig, renderer, *, w_pad=0, h_pad=0):
+    # Figure out how large the suptitle is and make the
+    # top level figure margin larger.
+    for panel in fig.panels:
+        _make_margin_suptitles(panel, renderer, w_pad=w_pad, h_pad=h_pad)
+    invTransFig = fig.transPanel.inverted().transform_bbox
+
+    w_pad, h_pad = (fig.transPanel - fig.transFigure).transform((w_pad, h_pad))
+
+    if fig._suptitle is not None and fig._suptitle.get_in_layout():
+        bbox = invTransFig(fig._suptitle.get_tightbbox(renderer))
+        p = fig._suptitle.get_position()
+        fig._suptitle.set_position((p[0], 1-h_pad))
```

Comment:
Sorry, can you clarify?  I'm not following the sequence of events you are thinking about here...

Reply:
`do_constrained_layout` calls `_make_margin_suptitles`, which modifies suptitle position here, then at the end of `do_constrained_layout`, the `_check_no_collapsed_axes` returns `False`, so `Axes` are not repositioned, but the suptitle has already been moved?

Reply:
Oh, I see - sorry the link wasn't correct.  

I *could* split this up - but its a sort-of fussy complication to do so, because the suptitle is just placed at the top of the figure anyway with the only difference being a slightly different space between the top and the title.